ts/haskell/mod.ts

```diff
@@ -1,35 +1,72 @@
-import { packageToEnvironment, shell } from "../environment.ts";
+import { packageToEnvironment } from "../environment.ts";
 import { mkPackage, Package } from "../package.ts";
 import { mkProject, Project } from "../project.ts";
+import { Executable } from "../executable.ts";
 import { nixSource } from "../internal/utils.ts";
-import { nixRaw, nixStrLit } from "../nix.ts";
+import { nixRaw } from "../nix.ts";
+
+/**
+ * Project components returned by `mkHaskellProject`
+ */
+export type HaskellAddenda = {
+  /**
+   * A package building the entire Haskell project
+   */
+  pkg: Package;
+  /**
+   * Make an executable from the cabal file available to garn.
+   */
+  addCabalExecutable: <T extends Project & HaskellAddenda, Name extends string>(
```

Comment:
Maybe we want to only require `ProjectData` here in case some of the project helper methods are overridden?

```suggestion
  addCabalExecutable: <T extends ProjectData & HaskellAddenda, Name extends string>(
```

Reply:
I tried this a bit more, and the problem is, as you suggested, that then we end up with more code duplication in addExecutable, which I'd have to essentially inline.